File: GMI-->Angular 2/src/app/Models/Chaudiere.ts
export class Chaudiere {
    id!: number;
    modele!: string;
    numeroSerie!: string;
    dateInstallation!: Date; // Use ISO string format for dates
    dateDerniereMaintenance!: Date; // Use ISO string format for dates
    description!: string; // New field for description
  }
  

File: Générale Maintenance Industrielle/src/main/java/com/alibou/security/Entities/Chaudiere.java
package com.alibou.security.Entities;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Chaudiere implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String modele;
    private String numeroSerie;
    private LocalDate dateInstallation;
    private LocalDate dateDerniereMaintenance;


    private String description;

}
